server/services/licenceService.ts

```diff
@@ -11,63 +11,78 @@ import {
   LicenceApiTestData,
 } from '../@types/licenceApiClientTypes'
 import LicenceApiClient from '../data/licenceApiClient'
-import { getStandardConditions } from '../utils/conditionsProvider'
-import { simpleDateTimeToJson, addressObjectToString } from '../utils/utils'
+import { getStandardConditions, getVersion } from '../utils/conditionsProvider'
+import { simpleDateTimeToJson, addressObjectToString, convertToTitleCase, convertDateFormat } from '../utils/utils'
 import PersonName from '../routes/creatingLicences/types/personName'
 import SimpleDateTime from '../routes/creatingLicences/types/simpleDateTime'
 import Telephone from '../routes/creatingLicences/types/telephone'
 import Address from '../routes/creatingLicences/types/address'
 import BespokeConditions from '../routes/creatingLicences/types/bespokeConditions'
-import PrisonRegisterApiClient from '../data/prisonRegisterApiClient'
 import LicenceStatus from '../enumeration/licenceStatus'
+import PrisonerService from './prisonerService'
+import CommunityService from './communityService'
 
 export default class LicenceService {
-  constructor(private readonly hmppsAuthClient: HmppsAuthClient) {}
+  constructor(
+    private readonly hmppsAuthClient: HmppsAuthClient,
+    private readonly prisonerService: PrisonerService,
+    private readonly communityService: CommunityService
+  ) {}
 
   async getTestData(username: string): Promise<LicenceApiTestData[]> {
     const token = await this.hmppsAuthClient.getSystemClientToken(username)
     return new LicenceApiClient(token).getTestData()
   }
 
-  async createLicence(username: string): Promise<LicenceSummary> {
-    const token = await this.hmppsAuthClient.getSystemClientToken(username)
+  async createLicence(prisonerNumber: string, username: string): Promise<LicenceSummary> {
+    const staffDetail = await this.communityService.getStaffDetail(username)
+    const nomisRecord = await this.prisonerService.getPrisonerDetail(username, prisonerNumber)
+    const deliusRecord = await this.communityService.getProbationer(prisonerNumber)
+    const prisonInformation = await this.prisonerService.getPrisonInformation(username, nomisRecord.agencyId)
 
```

Comment:
Could call these in parallel to save on sequential wait times? 
e.g.
`const [staffDetail, nomisRecord, deliusRecord, prisonInformation = await Promise.all([
   this.communityService.getStaffDetail(username),
   this.prisonerService.getPrisonerDetail(username, prisonerNumber),
   this.communityService.getProbationer(prisonerNumber),
   this.prisonerService.getPrisonInformation(username, nomisRecord.agencyId),
]`